pkg/inject/report_test.go

```diff
@@ -0,0 +1,145 @@
+		expected            bool
+	}{
+		{
+			podSpec:  &corev1.PodSpec{HostNetwork: false},
+			podMeta:  &metav1.ObjectMeta{},
+			expected: true,
+		},
+		{
+			podSpec:  &corev1.PodSpec{HostNetwork: true},
+			podMeta:  &metav1.ObjectMeta{},
+			expected: false,
+		},
+		{
+			podSpec: &corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  k8s.ProxyContainerName,
+						Image: "gcr.io/linkerd-io/proxy:",
+					},
+				},
+			},
+			podMeta:  &metav1.ObjectMeta{},
+			expected: false,
+		},
+		{
+			podSpec: &corev1.PodSpec{
+				InitContainers: []corev1.Container{
+					{
+						Name:  k8s.InitContainerName,
+						Image: "gcr.io/linkerd-io/proxy-init:",
+					},
+				},
+			},
+			podMeta:  &metav1.ObjectMeta{},
+			expected: false,
+		},
+		{
+			unsupportedResource: true,
+			podSpec:             &corev1.PodSpec{},
+			podMeta:             &metav1.ObjectMeta{},
+			expected:            false,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+				},
+			},
+			expected: true,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectDisabled,
+				},
+			},
+			expected: false,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+				},
+			},
+			nsAnnotations: map[string]string{
+				k8s.ProxyInjectAnnotation: k8s.ProxyInjectDisabled,
+			},
+			expected: true,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectDisabled,
+				},
+			},
+			nsAnnotations: map[string]string{
+				k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+			},
+			expected: false,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectDisabled,
+				},
+			},
+			nsAnnotations: map[string]string{
+				k8s.ProxyInjectAnnotation: k8s.ProxyInjectDisabled,
+			},
+			expected: false,
+		},
+		{
+			podSpec: &corev1.PodSpec{},
+			podMeta: &metav1.ObjectMeta{
+				Annotations: map[string]string{
+					k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+				},
+			},
+			nsAnnotations: map[string]string{
+				k8s.ProxyInjectAnnotation: k8s.ProxyInjectEnabled,
+			},
+			expected: true,
+		},
+	}
+
+	for i, testCase := range testCases {
+		testCase := testCase
+		t.Run(fmt.Sprintf("test case #%d", i), func(t *testing.T) {
+			resourceConfig := &ResourceConfig{}
+			resourceConfig.WithNsAnnotations(testCase.nsAnnotations)
+			resourceConfig.pod.spec = testCase.podSpec
+			resourceConfig.pod.meta = testCase.podMeta
+
+			report := newReport(resourceConfig)
+			report.update(resourceConfig)
```

Comment:
i think you can remove this line, now that we're calling `update` in `newReport`?